docs/advanced-querying-techniques/ctes.md

```diff
@@ -0,0 +1,398 @@
+
+```{code-cell} ipython3
+%%sql
+CREATE OR REPLACE TABLE bank AS
+FROM read_csv_auto('bank_cleaned.csv', header=True, sep=',')
+```
+
+We confirm the table was loaded
+
+```{code-cell} ipython3
+%sqlcmd tables
+```
+
+We can use [JupySQL's Table Explorer](https://jupysql.ploomber.io/en/latest/user-guide/table_explorer.html) to take a quick look at the table.
+
+```{code-cell} ipython3
+%sqlcmd explore --table bank
+```
+
+## Simple CTE
+
+Let's create a CTE that finds the average balance for each job type.
+
+```{code-cell} ipython3
+%%sql
+WITH Job_Avg_Balance AS (
+    SELECT job, AVG(balance) AS AverageBalance
+    FROM bank
+    GROUP BY job
+)
+SELECT * 
+FROM Job_Avg_Balance;
+```
+
+In this example, the `WITH` keyword indicates the start of the CTE, which we call `Job_Avg_Balance`.
+
+Within the `Job_Avg_Balance` CTE, we form a query that returns the average balance for each job by computing the average of `balance` and grouping by `job`. 
+
+After closing the `WITH` statement, we then select all columns from the `Job_Avg_Balance` CTE. This returns two columns: `job` and `AverageBalance`. The final `SELECT` statement then retrieves the data from the CTE.
+
+## Multiple CTEs
+
+You can use multiple CTEs in a single query. Let's find the average balance per job type and average campaign per job type.
+
+```{code-cell} ipython3
+%%sql
+WITH Job_Avg_Balance AS (
+    SELECT job, AVG(balance) AS AverageBalance
+    FROM bank
+    GROUP BY job
+),
+Job_Avg_Campaign AS (
+    SELECT job, AVG(campaign) AS AverageCampaign
+    FROM bank
+    GROUP BY job
+)
+SELECT * 
+FROM Job_Avg_Balance, Job_Avg_Campaign 
+WHERE Job_Avg_Balance.job = Job_Avg_Campaign.job;
+```
+
+In this example, the first CTE is the same as in the previous example. The second CTE `Job_Avg_Campaign` returns two columns: `job_1` and `AverageCampaign`. The final SELECT statement retrieves data from both CTEs. Notice that the entries.
+
+You will notice `job` and `job_1` in the final result. This is because we are doing a Cartesian product (cross join) between two CTEs and both have a column named `job`. We can avoid this by explicitly specifying the columns you want to select in our final `SELECT` statement instead of using `SELECT *`.
+
+```{code-cell} ipython3
+%%sql
+WITH Job_Avg_Balance AS (
+    SELECT job, AVG(balance) AS AverageBalance
+    FROM bank
+    GROUP BY job
+),
+Job_Avg_Campaign AS (
+    SELECT job, AVG(campaign) AS AverageCampaign
+    FROM bank
+    GROUP BY job
+)
+SELECT Job_Avg_Balance.job, AverageBalance, AverageCampaign
+FROM Job_Avg_Balance
+JOIN Job_Avg_Campaign 
+ON Job_Avg_Balance.job = Job_Avg_Campaign.job;
+```
+
+## Recursive CTEs
+
+A recursive CTE is one in which an initial CTE is repeatedly executed to return subsets of data until the complete result set is obtained.
+
+The given dataset doesn't lend itself to a recursive CTE, as these are generally used for hierarchical or recursive data problems, which the bank dataset does not present.
```

Comment:
Can we add some real-world usecases of recursive CTEs and give examples of datasets (with links)

Reply:
I meant can we add the example datasets (references). not to replace the original dataset with new ones

Reply:
Can I resolve this?

Reply:
yes, please
currently we're focused on introducing ETL 
also, Eduardo asked we don't invest as much time on this as we prepare for other topics